### ISSUE TITLE
`Ambiguous class resolution` warnings when running `composer update`

### ISSUE THREAD

**Issue by adevade** [thephpleague/flysystem#1664]
### Bug Report

| Q                 | A       |
|-------------------|---------|
| Flysystem Version | 3.15.0   |
| Adapter Name      | local |
| Adapter version   | 3.15.0   |

#### Summary

I get warnings when Composer is generating the autoload files.

#### How to reproduce

Install a new Laravel app (`laravel new test` or `composer create-project laravel/laravel test`) and see the warning messages.

```
Generating optimized autoload files
Warning: Ambiguous class resolution, "League\Flysystem\Local\FallbackMimeTypeDetector" was found in both "[redacted]/vendor/league/flysystem-local\FallbackMimeTypeDetector.php" and "[redacted]/vendor/league/flysystem/src\Local\FallbackMimeTypeDetector.php", the first will be used.
Warning: Ambiguous class resolution, "League\Flysystem\Local\LocalFilesystemAdapter" was found in both "[redacted]/vendor/league/flysystem-local\LocalFilesystemAdapter.php" and "[redacted]/vendor/league/flysystem/src\Local\LocalFilesystemAdapter.php", the first will be used.
```

**Comment by whoami15** [comment]
I got the same.

**Comment by frankdejonge** [comment]
This is fixed in 3.15.1, thanks for reporting it.
